crates/optimism/node/tests/e2e/p2p.rs

```diff
@@ -9,22 +9,20 @@ async fn can_sync() -> eyre::Result<()> {
 
     let chain_id: ChainId = BASE_MAINNET.chain.into();
 
-    let (mut nodes, _tasks, _wallet) = setup(2).await?;
+    let (mut nodes, _tasks) = setup(2).await?;
 
     let second_node = nodes.pop().unwrap();
     let mut first_node = nodes.pop().unwrap();
 
     let tip: usize = 300;
     let tip_index: usize = tip - 1;
 
-    let wallet = Wallet::default();
+    let wallet = WalletGenerator::new(1).chain_id(chain_id).gen().pop().unwrap();
```

Comment:
too verbose imo, if setup has chain_id, why not let it create a wallet as well?

Reply:
Hi Josh, sorry it is still WIP. Got some new commits and changed this a bit. I am still missing a final part tho.